.github/workflows/pr.yml

```diff
@@ -13,15 +28,19 @@ jobs:
       - name: 🚚 Get latest code
         uses: actions/checkout@v3
         with:
-          ref: ${{ github.head_ref }}
-          fetch-depth: 50
+          # default ref = ${{ github.ref }} = pr_merge_ref (a merge commit)
+          # Get PR's head commit
+          ref: ${{ env.PR_HEAD_SHA }}
+          fetch-depth: 0
 
-      - name: 🔧 Run eslint
+      - name: Install deps and git fetch
         run: |
           npm install
           git fetch --all
+
+      - name: 🔧 Run eslint on changed files only
           DIFF=$(git diff --name-only --diff-filter=ACMRT \
-            origin/${{ github.base_ref }}...origin/${{ github.head_ref }})
+            ${{ env.BASE_SHA }}...${{ env.PR_HEAD_SHA }})
```

Comment:
since we're touching this file, `::set-output` (line 30) is deprecated, and instead `echo "some-string-output" >> $GITHUB_OUTPUT` is preferred. If you can make that change, let me know. If not, that's okay.

https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/